packages/bot/src/scratch/index.js

```diff
@@ -74,15 +74,24 @@ export const scratchWorkspaceInit = async () => {
 let interpreter = null;
 
 export const runBot = (limitations = {}) => {
-    // Disable blocks outside of any main or independent blocks.
-    const top_blocks = Blockly.derivWorkspace.getTopBlocks();
-    top_blocks.forEach(block => {
-        if (!block.isMainBlock() && !block.isIndependentBlock()) {
-            block.setDisabled(true);
+    return new Promise((resolve, reject) => {
+        // Disable blocks outside of any main or independent blocks.
+        const top_blocks = Blockly.derivWorkspace.getTopBlocks();
+        top_blocks.forEach(block => {
+            if (!block.isMainBlock() && !block.isIndependentBlock()) {
+                block.setDisabled(true);
+            }
+        });
+
+        const { mainBlocks } = config;
+        const has_main_blocks = mainBlocks
+            .filter(block => block !== mainBlocks[2])
```

Comment:
- We could make this more clear by creating another entry in the constants file with blocks that are always required in a strategy.